pkg/tunneldriver/driver.go

```diff
@@ -480,6 +480,8 @@ func (td *TunnelDriver) CreateAgentEndpoint(ctx context.Context, name string, sp
 	if err != nil {
 		return err
 	}
+	td.agentEndpoints.Add(name, tun)
+	td.tunnels[name] = tun
```

Comment:
I don't think we need to add this to the tunnels map. The agent endpoints should be tracked and removed separately from tunnels.